### DISCUSSION TITLE
How to tune simply the wake parameters of curl model

### DISCUSSION THREAD

**Discussion by cjp606** [NatLabRockies/floris#389] [Q&A] [answered]
Hello everyone,

Recently, I was studying the wake of floating wind turbines, but I found that the wake speed loss of floating wind turbines is relatively small, and the wake expansion is more serious. Therefore, I want to add a coefficient to the velocity loss and expansion of the original curl wake, so that the wake model can better describe the wake characteristics of floating wind turbines, but because I am not familiar with FLORIS, I hope It is possible to know where simple modifications can be made to these two parameters of the wake model.

Thanks in advance.
Jiaping.Cui

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi Jiaping,

The input parameters are provided as a `.YAML` file to FLORIS. If you would like to modify certain parameters in the models, simply change their values in the `cc.yaml` input file. Input `.YAML` files are exemplified in the examples/inputs folder, and every Python code in the /examples/ folder will demonstrate how you load and then use FLORIS in various settings.

**Reply by cjp606** [reply]
Thank you vry much for your answer. It's helpful to solve my problem!

**Reply by cjp606** [reply]
Dear Bart,

About your method to change the parameters in a .yaml file, I found that it is a 3.0 or 3.0.1 version. So I try to run the V3.0, and it always diaplaied  errors as: 
"
Traceback (most recent call last):
  File "E:/FlorisPractice/FlorisPractice3/floris-3.0.1/examples/01_opening_floris_computing_power.py", line 19, in <module>
    from floris.tools import FlorisInterface
  File "F:\anaconda\anzhuangchu\lib\site-packages\floris\tools\__init__.py", line 39, in <module>
    from .floris_interface import FlorisInterface, FlorisInterface_legacy_v24
  File "F:\anaconda\anzhuangchu\lib\site-packages\floris\tools\floris_interface.py", line 32, in <module>
    from floris.utilities import Vec3
  File "F:\anaconda\anzhuangchu\lib\site-packages\floris\utilities.py", line 17, in <module>
    from attrs import define, field
ModuleNotFoundError: No module named 'attrs'
"
And I can not solve this problem with my most effort. Can you give me some suggestion?

Because I always use the V2.4.0 and this version can run normally, so is the path /examples/example_inputs.json if I want to change the parameters of the curl model?

Hoping to get your reply. Thank you in advance.
Jiaping.Cui

**Reply by rafmudaf** [reply]
@cjp606 If you want to upgrade to the latest version, see the installation instructions [here](https://nrel.github.io/floris/#installation)

**Reply by Bartdoekemeijer** [reply]
@cjp606 I have actually ran into similar issues. I solved the `attrs` problem by reinstalling it using `pip`:
```
pip uninstall attrs
pip install attrs
```

**Reply by cjp606** [reply]
Dear Rafael and Bart,

Thank you for your help, and I have solved the problem about the module "attrs", but it occured some other problem as fellows:

"Traceback (most recent call last):
  File "E:/FlorisPractice/FlorisPractice3/floris-3.0.1/examples/08_opt_yaw_single_ws.py", line 18, in <module>
    from floris.tools.optimization.yaw_optimization.yaw_optimizer_sr import (
  File "F:\anaconda\anzhuangchu\lib\site-packages\floris\tools\optimization\__init__.py", line 1, in <module>
    from . import other, scipy, pyoptsparse, yaw_optimization
ImportError: cannot import name 'yaw_optimization' from partially initialized module 'floris.tools.optimization' (most likely due to a circular import) (F:\anaconda\anzhuangchu\lib\site-packages\floris\tools\optimization\__init__.py)
"

I don't know how to solve this troblem, Hoping to get your reply.

Thanks in advance.

**Reply by Bartdoekemeijer** [reply]
Hi @cjp606 . This may have to do with Issue #360 . Can you try updating `floris` and reinstalling it?
